src/models/dish.ts

```diff
@@ -0,0 +1,63 @@
+import {Document, Schema, model, Types} from "mongoose";
+
+export interface IDish extends Document{
+    name: string;
+    cooker: Types.ObjectId;
+    description: string;
+    images: string[];
+    quantity: number;
+    price: number;
+    category: "vegeterian" | "non-vegeterian" | "gluten-free" | "vegan";
+    sold? : boolean;
+    slug? : string;
+    ratingsAvagare?: number;
+    ratingsQuantity?: number
+}
+
+const DishSchema = new Schema<IDish>({
+    name: {
+        type: String,
+        required: true
+    },
+    cooker: {
+        type: Schema.Types.ObjectId, 
+        ref: 'User'
+    },
+    description: {
+        type: String,
+        required: true
+    },
```

Comment:
i think it would be better if we added more validation, we will create a validator layer later on for specified checks.

` description: {
      type: String,
      required: [true, 'Product description is required'],
      minlength: [20, 'Too short dish description'],
    },`